### PR TITLE
Fix flaky test

### PR DESCRIPTION
**Pull request by mitchellhenke**
This test sometimes fails because the arrays have the same elements in a different order

### CODE REVIEW

**Review by aduth — APPROVED** [review]
:+1:

I thought this might be a problem since the method [does expect a sort order](https://github.com/18F/identity-dashboard/blob/5460756ea8520ebbd63dac865f475b1298d531e9/app/models/user.rb#L21-L25), but I see there's a test case below which guarantees this.

I imagine the flakiness could be something in [how we generate service providers](https://github.com/18F/identity-dashboard/blob/5460756ea8520ebbd63dac865f475b1298d531e9/spec/factories/service_providers.rb#L3), e.g. `["test98","test99","test100"].sort == ["test100", "test98", "test99"]`.

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -33,7 +33,8 @@
       user_sp = create(:service_provider, user: user)
       team_sp = create(:service_provider, team: team)
       create(:service_provider)
-      expect(user.scoped_service_providers).to eq([user_sp, team_sp])
+      sorted_sps = [user_sp, team_sp].sort_by { |x| x.friendly_name.downcase }
+      expect(user.scoped_service_providers).to eq(sorted_sps)
```
Comment (zachmargolis) [inline spec/models/user_spec.rb:37]: We can use RSpec's `match_array` to simplify this (it does order-independent comparison)

```suggestion
      expect(user.scoped_service_providers).to match_array([user_sp, team_sp])
```

some examples: https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers

File: yarn.lock [hunk 2]
```diff
@@ -900,17 +900,11 @@
...
-  resolved "https://registry.yarnpkg.com/@types/events/-/events-3.0.0.tgz#2862f3f58a9a7f7c3e78d79f130dd4d71c25c2a7"
-  integrity sha512-EaObqwIvayI5a8dCzhFrjKzVwKLxjoG9T6Ppd5CEo07LRKfQ8Yokw54r5+Wq7FaBQ+yXRvQAYPrHwya1/UFt9g==
-
 "@types/glob@^7.1.1":
-  version "7.1.1"
-  resolved "https://registry.yarnpkg.com/@types/glob/-/glob-7.1.1.tgz#aa59a1c6e3fbc421e07ccd31a944c30eba521575"
-  integrity sha512-1Bh06cbWJUHMC97acuD6UMG29nMt0Aqz1vF3guLfG+kHHJhy3AyohZFFxYk2f7Q1SQIrNwvncxAE0N/9s70F2w==
+  version "7.1.3"
+  resolved "https://registry.yarnpkg.com/@types/glob/-/glob-7.1.3.tgz#e6ba80f36b7daad2c685acd9266382e68985c183"
+  integrity sha512-SEYeGAIQIQX8NN6LDKprLjbrd5dARM5EXsd8GI/A5l0apYI1fGMWgPHSe4ZKL4eozlAyI+doUE9XbYS4xCkQ1w==
```
Comment (zachmargolis) [inline yarn.lock:906]: Do we need all these yarn updates too?

**Review by zachmargolis — APPROVED** [review]
LGTM, one suggestion

File: yarn.lock (hunk 2 above)
Comment (mitchellhenke) [reply yarn.lock:906]: I was trying to sneak in the vulnerability fixes 😬

File: spec/models/user_spec.rb (hunk 1 above)
Comment (mitchellhenke) [reply spec/models/user_spec.rb:37]: That's what I had initially, but Andrew caught that `scoped_service_providers` is actually sorted by `friendly_name`, so I changed the test to match that behavior

Comment (zachmargolis) [reply spec/models/user_spec.rb:37]: Oh ok! sounds good

**Pull request merged**